src/platform/ESP32/KeyValueStoreManagerImpl.cpp

```diff
@@ -29,12 +29,39 @@
 
 #include "nvs.h"
 #include "nvs_flash.h"
+#include <crypto/CHIPCryptoPAL.h>
 #include <lib/support/CodeUtils.h>
 
 namespace chip {
 namespace DeviceLayer {
 namespace PersistedStorage {
 
+namespace {
+// Implementation Note: esp-idf nvs implementation cannot handle key length > 15,
+// Below implementation tries to handle that case by hashing the key
+// If key length is > 15 then take the SHA1 of the key and convert the first 8 bytes to hex string.
+// Not sure how likely we would run into a conflict as we are only using 8 bytes out of 20
+
+CHIP_ERROR HashIfLongKey(const char * key, char * keyHash)
+{
+    VerifyOrReturnError(strlen(key) > (NVS_KEY_NAME_MAX_SIZE - 1), CHIP_ERROR_INCORRECT_STATE);
```

Comment:
Won't this return an error when writing a shorter key?

Reply:
The callers use `VerifyOrdo(HashIfLongKey() == CHIP_NO_ERROR)`, so that part is fine.  But seems like we could just return a boolean indicating whether the key was long....